packages/next/src/client/components/react-dev-overlay/hot-reloader-client.tsx

```diff
@@ -50,16 +51,37 @@ interface Dispatcher {
 let mostRecentCompilationHash: any = null
 let __nextDevClientId = Math.round(Math.random() * 100 + Date.now())
 let reloading = false
+let startLatency: number | null = null
 
 function onBeforeFastRefresh(dispatcher: Dispatcher, hasUpdates: boolean) {
   if (hasUpdates) {
     dispatcher.onBeforeRefresh()
   }
 }
 
-function onFastRefresh(dispatcher: Dispatcher, hasUpdates: boolean) {
+function onFastRefresh(
+  dispatcher: Dispatcher,
+  sendMessage: (message: string) => void,
+  updatedModules: string[]
+) {
+  let endLatency = Date.now()
```

Comment:
Should you use `performance.now()` instead: https://developer.mozilla.org/en-US/docs/Web/API/Performance/now?

Reply:
We need a pair of timestamps for creating spans, and unfortunately `performance.now()` is relative to navigation start.